CMakeLists.txt

```diff
@@ -1,28 +1,35 @@
 cmake_minimum_required(VERSION 3.2)
 project(trantor)
 
-if(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
-#Clang, use c++17
-    set(DEFS "USE_STD_ANY")
-    set(USE_STD_ANY 1)
-    set(CMAKE_CXX_STD_FLAGS c++17)
-else()
-    EXEC_PROGRAM (gcc ARGS "--version | grep '^gcc'|awk '{print $3}' | sed s'/)//g' | sed s'/-.*//g'" OUTPUT_VARIABLE version)
-    MESSAGE(STATUS "This is gcc version:: " ${version})
-    if(version LESS 5.4.0)
-        MESSAGE(STATUS "gcc is too old")
-        stop()
-    elseif(version LESS 7.1.0)
-        set(CMAKE_CXX_STD_FLAGS c++14)
-        MESSAGE(STATUS "c++14")
+include(CheckCXXCompilerFlag)
+CHECK_CXX_COMPILER_FLAG("-std=c++14" compiler_supports_cxx14)
+CHECK_CXX_COMPILER_FLAG("-std=c++17" compiler_supports_cxx17)
+
+if(NOT compiler_supports_cxx14)
+    message(FATAL_ERROR "Error: Your C++ compiler is too old for trantor")
+endif()
+
+if(NOT DEFINED CMAKE_CXX_STANDARD)
+    if(compiler_supports_cxx17)
+        include(CheckIncludeFileCXX)
+        CHECK_INCLUDE_FILE_CXX(string_view support_string_view)
+        if(support_string_view)
+            set(CMAKE_CXX_STANDARD 17)
+        else()
+            set(CMAKE_CXX_STANDARD 14)
+        endif()      
     else()
-        set(CMAKE_CXX_STD_FLAGS c++17)
-        set(DEFS "USE_STD_ANY")
-        set(USE_STD_ANY 1)
-        MESSAGE(STATUS "c++17")
+        set(CMAKE_CXX_STANDARD 14)
     endif()
 endif()
 
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+set(CMAKE_CXX_EXTENSIONS OFF)
+
+if(CMAKE_CXX_STANDARD LESS 14)
+    message(FATAL_ERROR "Error: c++14 or c++17 is required!")
+endif()
```

Comment:
I am afraid not, because I still need std::string_view, as belows:
https://github.com/an-tao/trantor/blob/931ada067572ed9e2ed5a2197267eec3c8c25d1a/trantor/utils/LogStream.h#L195-L204